testing-common/src/main/java/io/opentelemetry/instrumentation/testing/provider/MemoryLogRecordExporterComponentProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.testing.provider;
+
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ComponentProvider;
+import io.opentelemetry.sdk.logs.export.LogRecordExporter;
+import java.util.Objects;
+
+public class MemoryLogRecordExporterComponentProvider
```

Comment:
Perhaps it would be better to call it `Testing` instead of `Memory`?

Reply:
- aligns with https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/f87ae0b72896c14b042cc9cf435d2e0bb2de8dff/smoke-tests-otel-starter/spring-smoke-testing/src/main/java/io/opentelemetry/spring/smoketest/SpringSmokeOtelConfiguration.java#L24
- uses an InMemory..Exporter
- "test" could also refer to a setup that sends the data to a test sink using HTTP

Reply:
ok - will rename